metrics/config.go

```diff
@@ -103,19 +104,27 @@ type metricsConfig struct {
 	stackdriverProjectID string
 	// allowStackdriverCustomMetrics indicates whether it is allowed to send metrics to
 	// Stackdriver using "global" resource type and custom metric type if the
-	// metrics are not supported by "knative_revision" resource type. Setting this
+	// metrics are not supported by the registered monitored resource types. Setting this
 	// flag to "true" could cause extra Stackdriver charge.
 	// If backendDestination is not Stackdriver, this is ignored.
 	allowStackdriverCustomMetrics bool
+	// allowStackdriverCustomMetricsDomain indicates whether it is allowed to send metrics to
+	// Stackdriver using "global" resource type and custom metric type if the
+	// metrics are not supported by the registered monitored resource types. It allows to
+	// also include the configured metrics domain as part of the metric name. Setting this
+	// flag to "true" could cause extra Stackdriver charge.
+	// If backendDestination is not Stackdriver, this is ignored.
+	allowStackdriverCustomMetricsDomain bool
```

Comment:
Instead of introduces a bool flag to allow custom metrics domain, could we introduce a string called `subDomain` or something similar, then the full custom metric prefix is "custom.googleapis.com/<sub-domain>/<component>".

The subDomain defaults to `knative.dev/` at first. Then we use it in serving and make it as a requirement input.

Reply:
Makes sense!
So a new config variable named `metrics.stackdriver-subdomain` works? 
And if not specified, then we default to `knative.dev/`?

Reply:
`metrics.stackdriver-custom-metrics-domain` ?

Reply:
the name sounds good to me